presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -357,5 +348,12 @@ public void noMoreLifespans()
             pendingCompleted.clear();
             return result;
         }
+
+        @Override
+        public void notifyAllLifespansFinishedExecution()
+        {
+            checkState(scheduleCompleted);
```

Comment:
Why not delegate this call?

Reply:
@arhimondr : Underlying `SourceScheduler` is ungrouped execution.